src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -20,24 +26,26 @@
   parameters. In a GET request this is contained in various query
   parameters, for POST requests this should be contained in the body
   of the request"
-  {:query (s/maybe [s/Any])
-   :count? s/Bool
+  {(s/optional-key :query) (s/maybe [s/Any])
+   (s/optional-key :include_total) (s/maybe s/Bool)
    (s/optional-key :order_by) (s/maybe [[(s/one s/Keyword "field")
                                          (s/one (s/enum :ascending :descending) "order")]])
+   (s/optional-key :distinct_resources) (s/maybe s/Bool)
```

Comment:
actually that was a get with POST params. with a real POST it fails as expected.